Added bulk insert

Rap22tor opened the pull request:
#3 Added SetBulkAsync

mqudsi commented:
This looks fine to me. Adding the SQL statement to the `Operations` is weird because that creates a new cached command that expects to be used w/ just a different parameters binding, but you're actually resetting the `DbCommand.Text` each time - however we're still saving on creating a new `DbCommand` each time (and then needing to dispose of it).

Can you please add a new unit test class and method that at minimum checks that the inserted values are found on a subsequent lookup, and a method that verifies that a second call to `SetBulkAsync()` works OK (given that you're manipulating the command text)?

Rap22tor commented:
If you prefer that the statement does not get added to `Operation` I could change that. I tried to adapt to your existing code as much as possible and since `UseAsync()` expects an `Operation` parameter I thought it would be best if I added it there but honestly was just not quite sure how you'd like it.

I added the requested unittests, I hope these work well. Let me know if anything is missing. 😃

mqudsi commented:
> If you prefer that the statement does not get added to Operation I could change that.

No, you did it right. That's the only way to borrow a DbCommand and DbConnection from the pool with `UseAsync()`.

--

It all looks to me; I've just cleaned a few things up to conform with the project standards (removed trailing whitespace, changed git log messages to be imperative) and merged it in. Thank you very much for your contribution!
